pkg/auth/inject.go

```diff
@@ -33,6 +37,18 @@ func (m DependencyMap) Provide(dependencyName string, ctx context.Context, tConf
 			// from tConfig
 			PwMinLength: 6,
 		}
+	case "AuthPrincipalStore":
+		return principal.NewStore(
+			db.NewSQLBuilder("auth", tConfig.AppName),
+			db.NewSQLExecutor(ctx, "postgres", tConfig.DBConnectionStr),
+			logging.CreateLogger(ctx, "auth_principal"),
+		)
+	case "PasswordAuthProvider":
+		return password.NewProvider(
+			db.NewSQLBuilder("auth", tConfig.AppName),
+			db.NewSQLExecutor(ctx, "postgres", tConfig.DBConnectionStr),
+			logging.CreateLogger(ctx, "provider_password"),
+		)
```

Comment:
I cannot make sense on creating this two provider.

Seem putting them in AuthInfoProvider will be simpler without much sacrifice.

Reply:
In my mind, `AuthPrincipalStore` and `PasswordAuthProvider` should not combine with `AuthInfoStore` because `AuthInfoStore` is in the core library, which is meant to be imported by all gears, while the other two are in auth gear.

`AuthInfoStore` is the crud layer of `_core_user` (and maybe `_core_role`, `_core_user_role` later), while `AuthPrincipalStore` and `PasswordAuthProvider` are crud layer of `_auth_xxx`.

Reply:
While AuthPrincipalStore and PasswordAuthProvider may indeed be combined to one interface. I think I may revise a little bit on how to better do polymorphism with the `_auth_principal` and `_auth_provider_xxx`.